Makefile.dep

```diff
@@ -629,6 +629,20 @@ ifneq (,$(filter spiffs,$(USEMODULE)))
     USEMODULE += mtd
 endif
 
+ifneq (,$(filter fatfs_vfs,$(USEMODULE)))
+    USEPKG += fatfs
+    USEMODULE += vfs
+
+    ifeq ($(BOARD),native)
+      USEMODULE += fatfs_diskio_native
+      FATFS_DISKIO_NATIVE_DEFAULT_FILE ?= \"riot_fatfs_disk.img\"
+    else
+      USEMODULE += fatfs_diskio_sdcard_spi
+      USEMODULE += auto_init_storage
+    endif
```

Comment:
> 1: "that means a lot of duplication" and "seems to be cleaner" sounds somewhat contrary to me.

From a centralization vs. modularization standpoint it doesn't seem so contrary.

> 2: the fact that FatFs (for now) can only be used with either an image file on native or with sdcard_spi is a restriction that relates directly to the FatFs diskio layer and not to a specific board.
> 3: the restriction of 2 is likely to be removed in the future by adding a more dynamic diskio layer (e.g. think of using FAT formatted USB sticks and sdcards simultaneously). This would then lead to removing this duplicated code from all boards makefiles again.

I think that is something that basically specific to any file system IO, not just FatFS, so yeah another layer of abstraction in-between board and FS seems to be in order (but okay yes, that can be done later on).

> 4: IMO a dev who adds a new board (and its Makefile.dep) shouldn't have to think about "weird FatFS-internal" stuff.

No, but they are the only one that can say specifically for the board "Hey, file systems go there".